frontend/src/features/public-form/PublicFormProvider.tsx

```diff
@@ -100,6 +101,17 @@ export const PublicFormProvider = ({
   formId,
   children,
 }: PublicFormProviderProps): JSX.Element => {
+  // Scrub phone numbers and email addresses from browser logs.
+  // Phone numbers will always start with +
+  const removePIIData = (_key: string, value: string) => {
+    return JSON.stringify(value)
+      .replace(
+        /([a-zA-Z0-9+._-]+@[a-zA-Z0-9._-]+\.[a-zA-Z0-9_-]+)/g,
+        'email=REDACTED',
+      )
+      .replace(/\+\d+/g, 'number=REDACTED')
+  }
+
```

Comment:
Calling `stringify()` in the stringify replacer callback to do a full text replacement on the formatted json looks really weird (and it's dangerous in case we accidentally break the json structure). A typical replacer looks like this instead:
```suggestion
  const removePIIData = (_key: string, value: string) => (typeof value === 'string'
        ? value
            .replace(
                /([a-zA-Z0-9+._-]+@[a-zA-Z0-9._-]+\.[a-zA-Z0-9_-]+)/g,
                'email=REDACTED',
            )
            .replace(/\+\d+/g, 'number=REDACTED')
        : value
  )
```

Reply:
[27dc52d](https://github.com/opengovsg/FormSG/pull/5667/commits/27dc52d16c11a3efefb04e22e5639c892504d20f)

Reply:
as discussed - we redact the information when deep cloning the structure, so that DD is still able to parse it as a JSON object